src/main/java/core/basesyntax/service/impl/WriterServiceImpl.java

```diff
@@ -0,0 +1,19 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.service.WriterService;
+import java.io.FileWriter;
+import java.io.IOException;
+
+public class WriterServiceImpl implements WriterService {
+    @Override
+    public void writeToCsv(String generatedReport, String filePath) {
+        if (generatedReport == null || filePath == null || !filePath.endsWith("csv")) {
```

Comment:
Similarly as above please extract the checks to a method and use more descriptive messages.